docs/update-strategies.md

```diff
@@ -1,6 +1,7 @@
 ## This doc is still under development
 
-* `all` strategy uses `version` passed to options as a cache tag. When version changes, old version cache is removed and new files are downloaded. Of course if files has same name were not changed (304 status returned), then probably browser won't download them again and will just update cache.
-* `changed` strategy is more advanced than `all` or `hash`. To work properly it requires output assets to have unique names between compilations, e.g. _file's unique hash_ in its name. **With this strategy enabled, `index.html` file (or other files without dynamic name) should be placed in `main` section of the cache, otherwise they won't be revalidated**.
+* **`all`** strategy uses `version` passed to `options` as a cache tag. When version changes, old version cache is removed and new files are downloaded. Of course if files with the same name were not changed (HTTP status 304), the browser probably won't download them again and will just update cache.
+* **`changed`** strategy is more advanced than `all` or `hash`. `offline-plugin` will calculate the files' hashes on it's own, independant of `webpack`. Even if a file is renamed, if the hash is the same, the file isn't redownloaded, but rather just renamed. **With this strategy enabled, `index.html` (or other files without dynamic name) should be placed in `main` section of the cache, otherwise they won't be revalidated**.
```

Comment:
There is no `hash` option anymore. Old hash version could be emulated with `{ updateStrategy: 'all', version: '[hash]' }`. Anyway, the best strategy is `changed`, `all` is being kept only for situation when `changed` stops working by some reason (no need to write that in the docs, just saying).

Reply:
"independant of `webpack` hash calculation" might be better

Reply:
> **With this strategy enabled, `index.html` (or other files without dynamic name) should be placed in `main` section of the cache, otherwise they won't be revalidated**.

I'm not sure tbh. As per https://github.com/NekR/offline-plugin/blob/master/src/index.js#L334 it should be true, I kind of doubt it too. The thing was that you shouldn't add assets with static names ('/' or '/blah.html') to `additional`/`optional` because they could be downloaded at any time, rather than on SW install/update. I think this is going to be true until we start calculate hashed in SW too to verify that file is the same it meant to be.

Reply:
Fixed... Let me know if it's still wrong in any way. I did not change anything regarding static assets. Do you want me to clarify / word it in any other way?